src/custom-element.js

```diff
@@ -394,7 +405,14 @@ function createBaseAmpElementProto(win) {
 
     /** @private {!./base-element.BaseElement} */
     this.implementation_ = new Ctor(this);
-    this.implementation_.createdCallback();
+
+    /**
+     * An element always starts in a unupgraded state until it's added to DOM
+     * for the first time in which case it can be upgraded immediately or wait
+     * for script download or `upgradeCallback`.
+     * @private {!UpgradeState}
+     */
+    this.upgradeState_ = UpgradeState.NOT_UPGRADED;
```

Comment:
I have trouble finding where where elements change the state if they start out in non-stubbed state.